JASP-Engine/JASP/R/networkanalysis.R

```diff
@@ -15,1760 +15,1356 @@
+  
+  if (!options[["ready"]] || is.null(network[["network"]]) || mainContainer$getError()) { # make empty table
+    if (nVar > 0L) { # otherwise, a 1 by 1 table with a . is generated by default
+      # create a table of nVariables by nVariables
+      table$setExpectedSize(nVar)
+      table[["Variable"]] <- variables
+      
+    }
+  } else { # fill with results
+    
+    allNetworks <- network[["network"]]
+    TBcolumns <- data.frame(Variable = variables)
+    for (i in seq_len(nGraphs)) {
+      
+      toAdd <- allNetworks[[i]][["graph"]]
+      if (!options[["weightedNetwork"]])
+        toAdd <- sign(toAdd)
+      if (!options[["signedNetwork"]])
+        toAdd <- abs(toAdd)
+      
+      toAdd <- as.data.frame(toAdd)
+      names(toAdd) <- paste0(variables, i)
+      
+      TBcolumns <- cbind(TBcolumns, toAdd)
+    }
+    table$setData(TBcolumns)
+  }
+  mainContainer[["weightsTable"]] <- table
 }
 
-# functions for plots ----
-.networkAnalysisCentralityPlot <- function(network, options, perform, oldPlot = NULL) {
-
-	if (!is.null(oldPlot) && !identical(oldPlot[["data"]], ""))
-		return(oldPlot)
-
-	plot <- list(
-		title = "Centrality Plot",
-		width = options[["plotWidthCentrality"]],
-		height = options[["plotHeightCentrality"]],
-		custom = list(width = "plotWidthCentrality", height = "plotHeightCentrality"),
-		data = "",
-		status = "complete"
-	)
-
-	if (perform == "run" && !is.null(network)) {
-
-		wide <- network[["centrality"]]
-		wideDf <- Reduce(rbind, wide)
-		if (length(wide) > 1) {
-			wideDf[["type"]] <- rep(names(network[["centrality"]]), each = nrow(wideDf) / length(wide))
-			Long <- reshape2::melt(wideDf, id.vars = c("node", "type"))
-			colnames(Long)[3] <- "measure"
-			Long[["graph"]] <- Long[["type"]]
-			Long[["type"]] <- TRUE # options[["separateCentrality"]]
-		} else {
-			Long <- reshape2::melt(wideDf, id.vars = "node")
-			colnames(Long)[2] <- "measure"
-			Long[["graph"]] <- NA
-		}
-
-		if (options[["abbreviateLabels"]])
-			Long[["node"]] <- base::abbreviate(Long[["node"]], options[["abbreviateNoChars"]])
+.networkAnalysisBootstrapTable <- function(bootstrapContainer, options, position) {
+  
+  # a table to show before bootstrapping to give some feedback to the user.
+  # this used to give an ETA but with the progress bars that isn't really necessary anymore.
+  
+  bootstrapType <- options[["BootstrapType"]]
+  substr(bootstrapType, 1, 1) <- toupper(substr(bootstrapType, 1, 1)) # capitalize first letter
+  
+  table <- createJaspTable(title = "Bootstrap summary of Network", position = position)
+  table$addColumnInfo(name = "type",               title = "Type", type = "string")
+  table$addColumnInfo(name = "numberOfBootstraps", title = "Number of bootstraps", type = "integer")
+  table[["type"]]               <- bootstrapType
+  table[["numberOfBootstraps"]] <- options[["numberOfBootstraps"]]
+  return(table)
```

Comment:
I think you mean to assign this to the bootstrapContainer rather than returning it

Reply:
Good catch!